build_runner/test/generate/run_script_test.dart

```diff
@@ -0,0 +1,136 @@
+@Tags(['integration'])
+
+import 'dart:convert';
+
+import 'package:io/io.dart';
+import 'package:test/test.dart';
+import 'package:test_descriptor/test_descriptor.dart' as d;
+
+import 'package:_test_common/common.dart';
+
+main() {
+  group('run_script validation tests', () {
+    // The TestBuilder() will create a `*.copy` of
+    // whatever it is given, so we'll use it to create a
+    // copy of an executable Dart file (bin/main.dart).
+    //
+    // The expected output of running the generated file
+    // will be "it works!".
+    setUp(() async {
+      var executableFileContent = '''
+void main(List<String> args) {
+  if (args.isEmpty) {
+    print("it works!");
+  } else if (args[0] == "throw") {
+    throw StateError('oh no!');
+  } else {
+    print(args.join(';'));
+  }
+}
+      ''';
+      var originalBuildContent = '''
+import 'dart:io';
+import 'package:build_runner/build_runner.dart';
+import 'package:build_runner_core/build_runner_core.dart';
+import 'package:build_test/build_test.dart';
+
+main(List<String> args) async {
+  exitCode = await run(
+      args, [applyToRoot(new TestBuilder(
+        buildExtensions: {
+          '.dart': ['.copy.dart'],
+          '.txt': ['.txt.copy']
+        }
+      ))]);
+}
+''';
+
+      // Create the sandbox directory, including a build script,
+      // and the actual executable file.
+      await d.dir('a', [
+        await pubspec('a', currentIsolateDependencies: [
+          'build_runner',
+          'build_runner_core',
+          'build_test',
+          'glob'
+        ]),
+        d.dir('bin', [
+          d.file('main.dart', executableFileContent),
+          d.file('main.txt', 'cannot run this'),
+        ]),
+        d.dir('tool', [d.file('build.dart', originalBuildContent)]),
+      ]).create();
+
+      // Get the dependencies.
+      await pubGet('a');
+
+      // We don't need to run a build in the setUp() closure, because
+      // that's not the functionality we're testing.
+    });
+
+    test('at least one argument must be provided', () async {
+      // Should throw error 64.
+      var result = await runDart('a', 'tool/build.dart',
+          args: ['run', '--output', 'build']);
+      expect(result.exitCode, ExitCode.usage.code,
```

Comment:
It would be good to do a basic `contains` check on the stdout/stderr as well to differentiate these test cases and make sure we are getting the correct error messages.